config/config/src/readLocalConfig.ts

```diff
@@ -22,6 +22,12 @@ export async function readLocalConfig (prefix: string): Promise<LocalConfig> {
         try {
           config[envReplace(key, process.env)] = envReplace(val, process.env)
         } catch {}
+
+        // childConcurrency & workspaceConcurrency is integers
+        // https://github.com/pnpm/pnpm/issues/5075
+        if (/^[\d.]+$/.test(val)) {
+          config[key] = Number(val)
```

Comment:
this doesn't seem correct as the key and value may have env variables in them (as you can see at line 23 there is some additional processing). You should create some conversion function and use it on the value at line 23

Reply:
fixed